File: EnumerationAsFlagVb/FileAttributeEnum.vb
<Flags()>
Public Enum FileAttributeEnum
    System = 1              ' 1     00000001
    Read = 1 << 1           ' 2     00000010
    Write = 1 << 2          ' 4     00000100
    Executable = 1 << 3     ' 8     00001000
    Hidden = 1 << 4         ' 16    00010000
    Folder = 1 << 5         ' 32    00100000
End Enum

File: EnumerationAsFlags/FileAttributeEnum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EnumerationAsFlagsCs
{
    [Flags]
    enum FileAttributeEnum
    {
        System = 1, // 1
        Read = 1 << 1,  // 2
        Write = 1 << 2, // 4
        Executable = 1 << 3,    // 8
        Hidden = 1 << 4,    // 16
        Folder = 1 << 5,    // 32
    }
}